client_authentication_test.go

```diff
@@ -80,6 +80,10 @@ func TestAuthenticateClient(t *testing.T) {
 	barSecret, err := hasher.Hash(context.TODO(), []byte("bar"))
 	require.NoError(t, err)
 
+	// a secret containing literal characters that would be affected by double-URL-decoding.
+	percentSecret, err := hasher.Hash(context.TODO(), []byte("%66%6F%6F"))
```

Comment:
Can we add a special char here as well, just to make sure this works correctly? Something like `<` or `$` or `§`

Reply:
Done - for POST and basic auth